packages/myst-to-ipynb/src/index.ts

```diff
@@ -6,31 +6,43 @@ import type { PageFrontmatter } from 'myst-frontmatter';
 import { writeMd } from 'myst-to-md';
 import { select } from 'unist-util-select';
 
-function sourceToStringList(src: string): string[] {
-  const lines = src.split('\n').map((s) => `${s}\n`);
-  lines[lines.length - 1] = lines[lines.length - 1].trimEnd();
-  return lines;
+function markdownString(file: VFile, md_cells: Block[]) {
+  const md = writeMd(file, { type: 'root', children: md_cells }).result as string;
+  return {
+    cell_type: 'markdown',
+    metadata: {},
+    source: md,
+  };
 }
 
 export function writeIpynb(file: VFile, node: Root, frontmatter?: PageFrontmatter) {
-  const cells = (node.children as Block[]).map((block: Block) => {
+  const cells = [];
+  const md_cells: Block[] = [];
+
+  for (const block of node.children as Block[]) {
     if (block.type === 'block' && block.kind === 'notebook-code') {
+      if (md_cells.length != 0) {
+        cells.push(markdownString(file, md_cells));
+        md_cells.length = 0;
+      }
       const code = select('code', block) as Code;
-      return {
+      cells.push({
         cell_type: 'code',
         execution_count: null,
         metadata: {},
         outputs: [],
-        source: sourceToStringList(code.value),
-      };
+        source: code.value,
+      });
+    } else {
+      md_cells.push(block);
     }
-    const md = writeMd(file, { type: 'root', children: block.children as any }).result as string;
-    return {
-      cell_type: 'markdown',
-      metadata: {},
-      source: sourceToStringList(md),
-    };
-  });
+  }
+
+  if (md_cells.length != 0) {
+    cells.push(markdownString(file, md_cells));
```

Comment:
@kp992 this is a "policy" decision to fuse Markdown cells. Is there a particular reason for doing this? With the base PR #1882, if you ingest an `ipynb` file and export an `ipynb`, then you'll end up with the same number of cells (perhaps the frontmatter might be moved, though). 

If you have a use-case for fusing Markdown cells, I'd suggest adding that as an option!

Reply:
I still think these changes need reverting. Consider the following Markdown notebook
````markdown
---
kernelspec:
  name: python3
  display_name: Python 3
---

```{code-cell} python3
import sys
```

+++ { "kind": "notebook-content" }
Desc 1

+++ { "kind": "notebook-content" }

Desc 2

```{code-cell} python3
import os
```
````

It's the MD equivalent of having four cells: 
- code
- md
- md
- code

In this PR, that gets flattened to 
- code
- md
- code

(and the `myst-to-md` also writes the block separator `+++` inside the cell, which is not quite right).

Can you elaborate on why you made this particular change? Did the original PR fail to convert something properly?